geodiff/src/tableschema.cpp

```diff
@@ -15,3 +17,30 @@ bool TableSchema::hasPrimaryKey() const
   }
   return false;
 }
+
+void tableSchemaPostgresToSqlite( TableSchema &tbl )
+{
+  // SQLite does not really care much about column types
+  // but it does not like "geometry(X, Y)" type due to parentheses so let's convert
+  // that to a simple type
+
+  for ( size_t i = 0; i < tbl.columns.size(); ++i )
+  {
+    TableColumnInfo &col = tbl.columns[i];
+    if ( col.type.rfind( "geometry(", 0 ) == 0 )
+    {
+      col.type = col.geomType;
+    }
+  }
+}
+
+void tableSchemaConvert( const std::string &driverSrcName, const std::string &driverDstName, TableSchema &tbl )
```

Comment:
Table schema keeps original data type names - so that if you are using the same driver, you do not loose those type names, and type conversion is explicit when switching between drivers.